otter/grade/utils.py

```diff
@@ -79,30 +83,64 @@ def prune_images(force=False):
         print("Prune cancelled.")
 
 
-def merge_scores_to_df(scores: List[GradingResults]) -> pd.DataFrame:  
-    """  
-    Convert a list of ``GradingResults`` objects to a scores dataframe, including a row  
-    with the total point values for each question.  
+def get_points_possible_df(scores: List[GradingResults]) -> pd.DataFrame:
+    """
+    From the list of ``GradingResults`` objects try to find one that does not have a catastrophic
+    failure and extract the points-per-question into its own DataFrame. This becomes the
+    first row of final_grades.csv. If all the ``GradingResults`` objects  are catastrophic failures
+    then returns an empty DataFrame
 
-    Args:  
-        scores (``list[otter.test_files.GradingResults]``): the score objects to merge  
+    Args:
+        scores (``list[otter.test_files.GradingResults]``): the score objects to locate the points per question
 
-    Returns:  
-        ``pd.DataFrame``: the scores dataframe  
+    Returns:
+        ``pd.DataFrame``: the points per question dataframe; if all failures empty dataframe
+    """
+    gr_completed = [s for s in scores if not s.has_catastrophic_failure()]
+    pts_poss_df = pd.DataFrame()
+    if gr_completed:
+        pts_poss_dict = {t: [gr_completed[0].to_dict()[t]["possible"]] for t in gr_completed[0].to_dict()}
+        pts_poss_dict[SCORES_DICT_FILE_KEY] = POINTS_POSSIBLE_LABEL
+        pts_poss_dict[SCORES_DICT_PERCENT_CORRECT_KEY] = "--"
```

Comment:
using `--` instead of `NA` will prevent this column from being parsed as a float. Please use `NA` instead, and make sure that all other numeric columns you create in this PR use `NA` and not `--` as well.

Reply:
yes, I was trying to clean up what final_grades.csv looked like. 

I am fine with this but the non-values for failed notebooks as well as the points-possible row will be a bunch of "NA"s written to final_grades.csv

Reply:
Upon further consideration, the existing behavior for other analogous situations (which I'd like to keep) is to assign 0 points, which prevents nans from polluting calculations. I think it might be better, esp. now that there's a status column we can use to indicate errors, to put in 0s for all scores and percentages in this case.

Reply:
If we use the na_values attribute to read final_grades.csv in the test cases and the na_rep attribute for the to_csv I think things are cleaner. 
-  `pd.read_csv("test/final_grades.csv", na_values='--')`
- `output_df.to_csv(os.path.join(output_dir, "final_grades.csv"), na_rep='--', index=False)`

I am running the tests now.

final_grades.csv looks good:
![image](https://github.com/user-attachments/assets/1bcc78fc-3e2e-4936-b274-8734409ace2d)

Reply:
If we're just using `--` as `NaN` for an internal representation that is never surfaced to the user, I'm not sure what the point is? And if `--` is going into the output CSV file, that's not good -- we don't want to add any friction to users being able to parse the CSV (i.e. requiring users to remember to add `na_rep="--"` to `pd.read_csv`). 

Either way, I still think 0s are more appropriate than `NaN`s, since that's in line with the existing behavior of Otter.

Reply:
ok, I can do this. My hesitation is that if there is a timeout or some other error, we are writing 0's as a score as if it was graded and got a zero. But I get the parsing becomes an issue.

For the points-per-question row, the grading status would be "NA" which works.

Reply:
Actually, can we set the grading_status to "--" for the points-per-question row.

Here is what it looks like:
![image](https://github.com/user-attachments/assets/69cd1880-0afe-45a5-ba0a-735e3d28eb7e)

Reply:
I also set the percent_correct in the points-per-question-row to 1.0. Made more sense than 0 in my mind.

Reply:
Another thought -- I could set a ``short_message`` attribute on the Exception and TimeoutException that is more user-friendly for the grading_status column.

Reply:
If we want to surface e.g. the exception text in the CSV file, that's fine w/ me (although I think it should be a separate PR). We do something like [this](https://github.com/ucbds-infra/otter-grader/blob/master/otter/test_files/__init__.py#L416-L422) to surface the exception text in the CSV file.

Reply:
OK, I am going to leave this for now.